hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicSecurityConfig.java

```diff
@@ -121,22 +121,27 @@ public SecurityConfig setClientPolicyConfig(PermissionPolicyConfig policyConfig)
      */
     @Override
     public Set<PermissionConfig> getClientPermissionConfigs() {
-        return Collections.unmodifiableSet(securityService.getClientPermissionConfigs());
+        Set<PermissionConfig> permissionConfigs = securityService != null
+                ? securityService.getClientPermissionConfigs()
+                : staticSecurityConfig.getClientPermissionConfigs();
+        return Collections.unmodifiableSet(permissionConfigs);
     }
 
     /**
      *  Updates client permission configuration cluster-wide.
      */
     @Override
     public SecurityConfig setClientPermissionConfigs(Set<PermissionConfig> permissions) {
+        if (securityService == null) {
```

Comment:
Should an unsupported operation be thrown, or a NPE/IllegalArgumentException. For the latter we have the Preconditions utility class that makes this code a bit less verbose.